### PR TITLE
Sort file entries in .build file

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

**Comment by eksperimental** [comment]
Mimics files listed by `ls` command.

**Comment by josevalim** [comment]
It is a private file, so what benefit does sorting offer? If we have to sort, basic sorting should be enough.

**Comment by eksperimental** [comment]
Make it more natural for humans looking into it if you are looking for a file.

File: lib/ex_doc/formatter/html.ex [hunk 1]
```diff
@@ -48,7 +48,7 @@ defmodule ExDoc.Formatter.HTML do
         generate_list(nodes_map.modules, nodes_map, config) ++
         generate_list(nodes_map.tasks, nodes_map, config) ++ generate_index(config)
 
-    generate_build(all_files, build)
+    generate_build(Enum.sort_by(all_files, &{Path.dirname(&1), &1}), build)
```
Comment (josevalim) [inline lib/ex_doc/formatter/html.ex:51]:
```suggestion
    generate_build(Enum.sort(all_files), build)
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**